src/vm/comsynchronizable.cpp

```diff
@@ -1431,6 +1431,29 @@ BOOL QCALLTYPE ThreadNative::YieldThread()
     return ret;
 }
 
+FCIMPL1(Object*, ThreadNative::GetThreadDeserializationTracker, StackCrawlMark* stackMark)
+{
```

Comment:
What is the rationale for implementing this as FCall and not in managed code? Is it just because of the stack mark?

How will that work on CoreRT/ProjectN? The `[ReflectionBlocked]` attribute?

Reply:
As you're suggesting, the intent is to prevent reflection during deserialization from being able to get the thread's deserialization tracker. On CoreRT/ProjectN, private framework members aren't reflectable, so we can just have a simple thread-local object as long as it's private..